### ISSUE TITLE
Add a CHANGELOG.md?

### ISSUE THREAD

**Issue by eriknaslund** [bonfire-networks/iconify_ex#18]
I noticed that iconify_ex doesn't have a CHANGELOG.md.

I recently did the mistake of upgrading to 0.6 and suddenly got "No icon set found..." errors in `:prod`.
Eventually I realized 0.6 wasn't strictly backwards compatible - you needed to add `env: config_env()` as part of the config.

I think it'd be great if we could have a CHANGELOG.md so that backwards incompatible changes can be announced. This way I'd feel much more comfortable updating iconify_ex to newer versions.

**Comment by mayel** [comment]
Yeah good point, I do try and follow semver to indicate breaking changes, but definitely need a changelog too, I just wish there was a tool that showed me all the commits since last release to use as a draft changelog entry (I actually hacked something like that ages ago, may need to revisit it).

**Comment by eriknaslund** [comment]
Great to hear that you're following semver. A lot of projects don't really any more , so I've kinda stopped "trusting it", if you know what I mean :(.

I might misunderstand what you're trying to do, but if you want you could do it like this:
`git log --oneline <revision>..HEAD`, e.g. `git log --oneline cdf3120d..HEAD`,
to see all commits between two revisions.

If you use `git tag` to tag releases in git as you're pushing them to hex.pm, it'd become even easier since you can then just do `git log --oneline v0.6.1..HEAD`.

Another option is to simple have a "Version NEXT" on top of the changelog in the main branch, and mention that the stuff documented here is not released on hex.pm yet. This way you can add to the changelog as part of doing the actual work, and when "release day" comes and it's time to push to hex.pm, simply change "Version NEXT" to "Version 1.6.2" for example.
